### PR TITLE
Overview beta page for initial fleet view implementation

### PR DESCRIPTION
**Pull request by zlayne**
Related story: https://issues.redhat.com/browse/ACM-6549
Initial implementation for the new Overview page that will provide an entry point for eventual CCX metrics/data & user configurable searches.

Screenshot 2023-08-03 at 11 03 11 AM

### CODE REVIEW

File: frontend/src/lib/usePrometheusPoll.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    const tick = () => savedCallback.current && savedCallback.current()
+
+    tick() // Run first tick immediately.
+
+    if (pollDelay) {
+      // Only start interval if a delay is provided.
+      const id = setInterval(tick, pollDelay)
+      return () => clearInterval(id)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (KevinFCormier) [inline frontend/src/lib/usePrometheusPoll.ts:84]: Not needed here. Please remove.

File: frontend/src/routes/Home/Overview/OverviewPage.tsx [hunk 2]
```diff
@@ -631,80 +629,78 @@ export default function OverviewPage() {
...
+                colorScale={colorThemes.criticalSuccess}
+              />
+              <AcmDonutChart
+                title={t('Cluster issues')}
+                description={t('Overview of cluster issues')}
+                loading={!policyReportData}
+                data={policyReportData}
+                donutLabel={{
+                  title: `${clustersWithIssuesCount}`,
+                  subTitle: clustersWithIssuesCount === 1 ? t('Cluster with issues') : t('Clusters with issues'),
```
Comment (KevinFCormier) [inline frontend/src/routes/Home/Overview/OverviewPage.tsx:689]: You could pass a `count` interpolation variable and use plural support in the translation library.

File: frontend/src/routes/Home/Overview/overviewDataFunctions.tsx [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+  managedClusterInfos: ManagedClusterInfo[],
+  applicationList: Application[],
+  appSets: ApplicationSet[],
+  filteredOCPApps: Record<string, any>,
+  t: TFunction<string, undefined>
+) {
+  const kubernetesTypes = new Set()
+  const regions = new Set() // regions could be retrieved from Node labels?
+  filteredClusters.forEach((curr: Cluster) => {
+    kubernetesTypes.add(curr.labels?.vendor ?? 'Other')
```
Comment (KevinFCormier) [inline frontend/src/routes/Home/Overview/overviewDataFunctions.tsx:176]: Is the `Other` value only used for grouping?

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
Looks great. Just a few small comments and questions. Thanks.

File: frontend/src/routes/Home/Overview/overviewDataFunctions.tsx (hunk 3 above)
Comment (zlayne) [reply frontend/src/routes/Home/Overview/overviewDataFunctions.tsx:176]: Yes, this is how the current Overview page groups for Kube types & regions. If the cluster does not have a label for vendor or region it is classified as other.

**Comment by zlayne** [comment]
@KevinFCormier Thanks for the review. Updated the PR. Let me know if you have any other concerns.

**Comment by Randy424** [comment]
@zlayne Thanks for taking on the code smells.

**Review by Randy424 — APPROVED** [review]
/lgtm

**Pull request merged**